### PR TITLE
feat(rome_js_analyze): `useLiteralKeys` handle string literal properties

### LINKED ISSUE #4690: `useLiteralKeys` rule should cover string literal properties
### Discussed in https://github.com/rome/tools/discussions/4664

<div type='discussions-op-text'>

<sup>Originally posted by **Conaclos** July  7, 2023</sup>
The [`useLiteralKeys`](https://docs.rome.tools/lint/rules/useLiteralKeys/) lint rule reports code where static property access can be used instead of computed acces.

```diff
- a["b"]
+ a.b
```

This implements the EsLint [`dot-notation`](https://eslint.org/docs/latest/rules/dot-notation) rule.

The EsLint [`no-useless-computed-key`](https://eslint.org/docs/latest/rules/no-useless-computed-key) rule reports useless computed keys on an object literal:

```diff
{
-  ["b"]: 0,
-  ["1+1"]: 2,
-  ["f"]() {},
+  b: 0,
+  "1+1": 2,
+  f() {},
}
```

I suggest merging these two EsLint rule into [`useLiteralKeys`](https://docs.rome.tools/lint/rules/useLiteralKeys/).
The only minor inconsistency in the rule name is the simplification of `["1+1"]` to `"1+1"`.
The rule could also handle the following cases:

```diff
{
-  "b": 0,
- "f"() {},
+  b: 0,
+ f() {},
}
```

```diff
class C {
-  ["b"] = 0
-  "c" = 0
- ["f"]() {}
-  "g"() {}
+  b = 0
+  c = 0
+  f() {}
+  g() {}
}
```

This could also be done on interfaces and object types.

This is not a breaking change since `useLiteralKeys` is a nursery rule.
</div>

### PR DESCRIPTION
**Pull request by Conaclos**
## Summary

Fix #4690

## Test Plan

New test included.

### CODE REVIEW

File: crates/rome_js_analyze/src/analyzers/nursery/use_literal_keys.rs [hunk 1]
```diff
@@ -51,70 +51,49 @@ declare_rule! {
...
         };
-
         match inner_expression {
             AnyJsExpression::AnyJsLiteralExpression(
                 AnyJsLiteralExpression::JsStringLiteralExpression(string_literal),
             ) => {
                 let value = string_literal.inner_string_text().ok()?;
-                if can_convert_to_static_member(&value) {
+                // A computed property `["something"]` can always be simplified to a string literal "something".
+                if matches!(node, AnyJsMember::JsComputedMemberName(_)) || is_js_ident(&value) {
```
Comment (ematipico) [inline crates/rome_js_analyze/src/analyzers/nursery/use_literal_keys.rs:81]: Isn't it the same for 
```js
a[`something`]
```

**Review by ematipico — APPROVED** [review]
Shouldn't we cover template literals too?

File: crates/rome_js_analyze/src/analyzers/nursery/use_literal_keys.rs (hunk 1 above)
Comment (Conaclos) [reply crates/rome_js_analyze/src/analyzers/nursery/use_literal_keys.rs:81]: Yes, It is handled just below :)

Comment (Conaclos) [reply crates/rome_js_analyze/src/analyzers/nursery/use_literal_keys.rs:81]: By the way, I can add a comment and a test

**Comment by Conaclos** [comment]
> Shouldn't we cover template literals too?

I am not sure what you refer to, so I just add a note o be sure:

template string property are not syntactically valid in js:

```js
const a = {
  `prop`: 0, // syntax error
}
```

**Comment by ematipico** [comment]
Oh, it's already handled. Could you be more precise in the changelog, then? It should phrase "string literal properties inside interfaces and objects"

**Pull request merged**